src/SensorManager.php

```diff
@@ -45,31 +45,77 @@
  *
  * @internal
  */
-class SensorManager
+final class SensorManager
 {
-    private ?CacheEventSensor $cacheEventSensor;
+    /**
+     * @var (callable(CacheEvent): void)|null
+     */
+    public $cacheEventSensor;
```

Comment:
In our tests, we previously had to extend this class. It was painful and caused unexpected exceptions under the hood.

This new approach makes these properties public and accept a callable. In our tests, we can then swap out one sensor with a closure. It is much nicer.